### PR TITLE
Update docstrings

### PR DESCRIPTION
**Pull request by eliascarv**
Added docstring: `choose`.
Updated doctring: `Select`, `Reject`, `Filter` and `DropMissing`.

### CODE REVIEW

File: src/colspec.jl [hunk 1]
```diff
@@ -41,7 +41,36 @@ end
 """
 const ColSpec = Union{Vector{T},NTuple{N,T},Regex,Colon} where {N,T<:ColSelector}
 
-# choose column names using colspec
+"""
+    choose(colspec::ColSpec, names) -> Vector{Symbol}
+
+Choose column names using colspec.
+The `names` argument will be the value returned by the `Tables.columnnames` function 
+which can be of type `Vector{Symbol}` or of type `NTuple{N,Symbol}`.
```
Comment (juliohm) [inline src/colspec.jl:49]:
```suggestion
Choose column `names` using `colspec`.
```

File: src/transforms/filter.jl [hunk 2]
```diff
@@ -6,6 +6,24 @@
     Filter(function)
 
 Filters the table returning only the rows where the `function` returns true.
```
Comment (juliohm) [inline src/transforms/filter.jl:8]: `function` is a Julia keyword, we should rename this to `func` or `pred` for predicate.

File: src/colspec.jl [hunk 3]
```diff
@@ -41,7 +41,36 @@ end
...
+
+# Examples
```
Comment (juliohm) [inline src/colspec.jl:51]:
```suggestion
# Examples

```

File: src/transforms/filter.jl [hunk 4]
```diff
@@ -6,6 +6,24 @@
...
+
+# Examples
```
Comment (juliohm) [inline src/transforms/filter.jl:10]:
```suggestion
# Examples

```

File: src/transforms/filter.jl [hunk 5]
```diff
@@ -6,6 +6,24 @@
...
+```julia
+julia> table = (a = [1, missing, 2, 3], b = [missing, 1, 2, 3]);
+
+julia> T = Filter(row -> all(!ismissing, row));
+
+julia> newtable, cache = apply(T, table);
+
+julia> newtable
+(a = Union{Missing, Int64}[2, 3], b = Union{Missing, Int64}[2, 3])
+```
```
Comment (juliohm) [inline src/transforms/filter.jl:26]:
```suggestion
```

This note should be written as a small comment in a ## Notes section. I think we have examples of notes in this repo already. If not, take a look at the GeoStats.jl or Meshes.jl projects where we have a bunch of notes in docstrings.

Notes shouldn't be the first thing a user reads, these are very technical details that should only come up at the end for advanced users trying to grasp nuances.

File: src/transforms/filter.jl [hunk 6]
```diff
@@ -46,11 +64,33 @@ Drop all rows with missing values in table.
     DropMissing([col₁, col₂, ..., colₙ])
     DropMissing((col₁, col₂, ..., colₙ))
 
-Drop all rows with missing values in selects columns `col₁`, `col₂`, ..., `colₙ`.
+Drop all rows with missing values in selects columns `col₁`, `col₂`, ..., `colₙ`.
```
Comment (juliohm) [inline src/transforms/filter.jl:67]:
```suggestion
Drop all rows with missing values in selected columns `col₁`, `col₂`, ..., `colₙ`.  
```

File: src/transforms/filter.jl [hunk 7]
```diff
@@ -46,11 +64,33 @@ Drop all rows with missing values in table.
     DropMissing([col₁, col₂, ..., colₙ])
     DropMissing((col₁, col₂, ..., colₙ))
 
-Drop all rows with missing values in selects columns `col₁`, `col₂`, ..., `colₙ`.
+Drop all rows with missing values in selects columns `col₁`, `col₂`, ..., `colₙ`.  
+The `col` arguments must be the same type and the accepted types 
+for `col` arguments are: `Integer`, `Symbol` or `String`.
```
Comment (juliohm) [inline src/transforms/filter.jl:69]:
```suggestion
```

This is a technical detail that is not useful to end users. They will just try whatever combination and will figure out by themselves that it works with Symbol, String, etc.

File: src/transforms/filter.jl [hunk 8]
```diff
@@ -46,11 +64,33 @@ Drop all rows with missing values in table.
...
 
     DropMissing(regex)
 
 Drop all rows with missing values in columns that match with `regex`.
+
+# Examples
```
Comment (juliohm) [inline src/transforms/filter.jl:75]:
```suggestion
# Examples

```

File: src/transforms/filter.jl [hunk 9]
```diff
@@ -46,11 +64,33 @@ Drop all rows with missing values in table.
...
+```julia
+T = DropMissing()
+T = DropMissing("b", "c", "e")
+T = DropMissing([2, 3, 5])
+T = DropMissing((:b, :c, :e))
+T = DropMissing(r"[bce]]")
```
Comment (juliohm) [inline src/transforms/filter.jl:81]:
```suggestion
T = DropMissing(r"[bce]")
```

File: src/transforms/filter.jl [hunk 10]
```diff
@@ -46,11 +64,33 @@ Drop all rows with missing values in table.
...
+Note that columns affected by DropMissing will have their schema changed:
+```julia
+julia> table = (a = [1, missing, 2, 3], b = [missing, 1, 2, 3]);
+
+julia> T = DropMissing();
+
+julia> newtable, cache = apply(T, table);
+
+julia> newtable
+(a = [2, 3], b = [2, 3])
```
Comment (juliohm) [inline src/transforms/filter.jl:93]:
```suggestion
```

Again this is a note, not something that end users should read right away. If you feel that this information is important, please create a ## Notes section at the bottom of the docstring with bullet points.

File: src/transforms/select.jl [hunk 11]
```diff
@@ -47,11 +47,21 @@ Tables.materializer(t::TableSelection) =
     Select([col₁, col₂, ..., colₙ])
     Select((col₁, col₂, ..., colₙ))
     
-The transform that selects columns `col₁`, `col₂`, ..., `colₙ`.
+The transform that selects columns `col₁`, `col₂`, ..., `colₙ`.  
+The `col` arguments must be the same type and the accepted types 
+for `col` arguments are: `Integer`, `Symbol` or `String`.
```
Comment (juliohm) [inline src/transforms/select.jl:52]:
```suggestion
```

File: src/transforms/select.jl [hunk 12]
```diff
@@ -47,11 +47,21 @@ Tables.materializer(t::TableSelection) =
...
     
     Select(regex)
 
 Selects the columns that match with `regex`.
+
+# Examples
```
Comment (juliohm) [inline src/transforms/select.jl:58]:
```suggestion
# Examples

```

File: src/transforms/select.jl [hunk 13]
```diff
@@ -119,11 +129,21 @@ revert(::Select, newtable::TableSelection, cache) = newtable.table
     Reject([col₁, col₂, ..., colₙ])
     Reject((col₁, col₂, ..., colₙ))
 
-The transform that discards columns `col₁`, `col₂`, ..., `colₙ`.
+The transform that discards columns `col₁`, `col₂`, ..., `colₙ`.  
+The `col` arguments must be the same type and the accepted types 
+for `col` arguments are: `Integer`, `Symbol` or `String`.
```
Comment (juliohm) [inline src/transforms/select.jl:134]:
```suggestion
```

File: src/transforms/select.jl [hunk 14]
```diff
@@ -119,11 +129,21 @@ revert(::Select, newtable::TableSelection, cache) = newtable.table
...
 
     Reject(regex)
 
 Discards the columns that match with `regex`.
+
+# Examples
```
Comment (juliohm) [inline src/transforms/select.jl:140]:
```suggestion
# Examples

```

**Review by juliohm — CHANGES_REQUESTED** [review]
Review attached 👍🏽 Please do not reply with "OK" in every suggestion I made, just press thumbs up if you want to express agreement. Replies are useful when they add new information to the discussion or when they highlight an issue that was not considered before.

**Comment by eliascarv** [comment]
I apply your suggestions in [37fab84](https://github.com/JuliaML/TableTransforms.jl/pull/47/commits/37fab84ca3d550a5d4e92fafdd5804f3288fb232) @juliohm. Plese, check the new notes. Also check that the grammar is correct.

File: src/transforms/filter.jl [hunk 15]
```diff
@@ -3,9 +3,20 @@
...
+# Examples
+
+```julia
+T = Filter(row -> sum(row) > 10)
+T = Filter(row -> row.a == true && row.b < 30)
+```
+
+## Notes
+
+* The schema of the new table will be the same as the original table.
```
Comment (juliohm) [inline src/transforms/filter.jl:19]:
```suggestion
* The schema of the table is preserved by the transform.
```

File: src/transforms/filter.jl [hunk 16]
```diff
@@ -46,11 +57,26 @@ Drop all rows with missing values in table.
...
+T = DropMissing("b", "c", "e")
+T = DropMissing([2, 3, 5])
+T = DropMissing((:b, :c, :e))
+T = DropMissing(r"[bce]")
+```
+
+## Notes
+
+* If the eltype of column affected by `DropMissing` is `Union{Missing,T}`,
+the eltype of new column will be `T`.
```
Comment (juliohm) [inline src/transforms/filter.jl:79]:
```suggestion
* The transform can potentially alter the element type of columns from `Union{Missing,T}` to simply `T`.
```

**Review by juliohm — CHANGES_REQUESTED** [review]

File: src/transforms/filter.jl [hunk 17]
```diff
@@ -46,11 +57,25 @@ Drop all rows with missing values in table.
...
+T = DropMissing()
+T = DropMissing("b", "c", "e")
+T = DropMissing([2, 3, 5])
+T = DropMissing((:b, :c, :e))
+T = DropMissing(r"[bce]")
+```
+
+## Notes
+
+* The transform can potentially alter the element type of columns from `Union{Missing,T}` to simply `T`.
```
Comment (juliohm) [inline src/transforms/filter.jl:78]:
```suggestion
* The transform can alter the element type of columns from `Union{Missing,T}` to `T`.
```

**Review by juliohm — APPROVED** [review]

**Pull request merged**